data-prepper-plugins/aws-lambda/src/test/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorTest.java

```diff
@@ -541,16 +562,16 @@ public void testConvertLambdaResponseToEvent_ExpectException_when_request_respon
         when(invokeResponse.payload()).thenReturn(lambdaReponse);
         when(invokeResponse.statusCode()).thenReturn(200); // Success status code
 
-        int randomCount = (int) (Math.random() * 10);
-        List<Record<Event>> originalRecords = getSampleEventRecords(randomCount);
         Buffer buffer = new InMemoryBuffer(lambdaProcessorConfig.getBatchOptions().getKeyName());
         for (Record<Event> originalRecord : originalRecords) {
             buffer.addRecord(originalRecord);
         }
         // Act
-        assertThrows(RuntimeException.class, () -> localLambdaProcessor.convertLambdaResponseToEvent(buffer, invokeResponse),
-                "For Strict mode request and response size from lambda should match");
-
+        if (null != expectedException) {
```

Comment:
It's better not to have conditionals in tests. This can be flaky and hard to understand.

Can we split this into two tests with two different ArgumentsSources methods/providers?

Reply:
Oh in this case, it is basically like, is there an exception to assert or no. The check is only based on the arguments but not based on the logic. I think we do have such pattern else where.

Reply:
I think it is a pattern best avoided.

Reply:
But, we can keep for now.